docs/proposals/project-scoped-appset.md

```diff
@@ -0,0 +1,114 @@
+---
+title: Support project scoped ApplicationSet
+authors:
+- "@jorturfer"
+
+sponsors:
+- TBD
+
+reviewers:
+- TBD
+
+approvers:
+- TBD
+
+creation-date: 2023-05-17
+last-updated: 2023-05-17
+---
+
+# Neat Enhancement Idea
+
+This is a proposal to provide support for users to use ApplicationSets inside the scope of a single project.
+
+
+## Open Questions [optional]
+
+- Should we include global resources (resources that aren't scoped to any project) into the available resources for project scoped appset?
+
+## Summary
+
+ArgoCD users will be able to use ApplicationSets in the context of the projects where they have access by RBAC. From external usage point of view, each time when an ApplicationSet is created/updated, the controller will validate if the user has permission for doing it. From controller point of view, during the Applications generation, the controller will ensure that the used resources are part of the project 
+
+## Motivation
+
+We have transversal teams where they are the owners of the application in all the stages (code, infra, ux, design,...) and our products are deployed in a cluster per product. Apart from this, critical products are deployed into multiple regions, which means that a product could be in 3-5 different Kubernetes clusters (including different cloud providers). 
+In this use case, ApplicationSet is the perfect match for us because they can deploy an appset for each microservice and automatically they are deployed into all the clusters with the required configuration (also enabling disaster recovery scenarios in other providers/regions thanks to the generators)
+
+As a platform team, we provide to the teams the tooling like ArgoCD, Harbor, observability,... This means that we manage an ArgoCD installation shared for all the teams, managing their clusters from the same place (based on RBAC). We'd like to allow the usage of appsets for the teams, but due to the global scope of them, teams can deploy things into other teams clusters because appsets ignores any kind of limitation.
+
+### Goals
+
+Add support for deploying ApplicationSets inside a AppProject and limit the interaction with elements inside it. 
+This includes also the RBAC verification
+
+### Non-Goals
+
+This doesn't include any change in the UI/cli to make them user friendly
+
+## Proposal
+
+The proposal for achieving this goal is to add a "required" `project` field (read more below) into the ApplicationSet and based on it, limit the access to the resources during the Applications generation. 
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: ApplicationSet
+metadata:
+  name: cluster-addons
+  namespace: argocd
+spec:
+  project: sample-project # new field
```

Comment:
Would it make sense for `project` to instead to be a list of `projects`? An `Application.project` matching any of the `projects` in the list would be able to be generated, subject to RBAC checks.

Reply:
It could make sense, yes